packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -18,19 +17,23 @@ describe('SyntheticEvent', () => {
   var createEvent;
 
   beforeEach(() => {
-    // TODO: can we express this test with only public API?
-    SyntheticEvent = require('events/SyntheticEvent').default;
     React = require('react');
     ReactDOM = require('react-dom');
     ReactTestUtils = require('react-dom/test-utils');
 
-    createEvent = function(nativeEvent) {
-      var target = require('../getEventTarget').default(nativeEvent);
-      return SyntheticEvent.getPooled({}, '', nativeEvent, target);
+    createEvent = (nativeEvent, callback) => {
+      var instance;
+      var container = document.createElement('div');
+      ReactDOM.render(
+        <div ref={ref => (instance = ref)} onClick={callback} />,
+        container,
+      );
+      ReactTestUtils.SimulateNative.click(instance, nativeEvent);
```

Comment:
Unfortunately Simulate* APIs are not very useful for testing because they are sort of like private APIs too. If you look at https://github.com/facebook/react/pull/11365 you'll see we're trying to avoid them and dispatch native browser events on DOM nodes instead. This PR should too.